From ee7b7387616575a248dcdaf9a1ed2525d99ed390 Mon Sep 17 00:00:00 2001
From: Michal Marek <mmarek@suse.com>
Date: Mon, 26 Oct 2015 11:05:47 +0100
Subject: [PATCH] kabi: Hide rt6_* types from genksyms on ppc64le
References: bsc#951546
Patch-mainline: Never, kabi

After enabling CONFIG_IP_VS_IPV6, the kabi checker sees the definitions
of struct fib6_table, struct rt6_info and struct rt6_statistics.

Signed-off-by: Michal Marek <mmarek@suse.com>
---
 net/netfilter/ipvs/ip_vs_core.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/netfilter/ipvs/ip_vs_core.c b/net/netfilter/ipvs/ip_vs_core.c
index 3b0757874933..0d0144169350 100644
--- a/net/netfilter/ipvs/ip_vs_core.c
+++ b/net/netfilter/ipvs/ip_vs_core.c
@@ -47,10 +47,12 @@
 #include <linux/netfilter_ipv4.h>
 
 #ifdef CONFIG_IP_VS_IPV6
+#if !defined(__powerpc64__) || !defined(__GENKSYMS__)
 #include <net/ipv6.h>
 #include <linux/netfilter_ipv6.h>
 #include <net/ip6_route.h>
 #endif
+#endif
 
 #include <net/ip_vs.h>
 
-- 
2.1.4

